### ISSUE TITLE
Bayesian GAM model knot specification

### ISSUE THREAD

**Issue by tbmurphy** [paul-buerkner/brms#1052]
I was using the brm function to fit a Bayesian GAM and everything worked perfectly. However, when I specified the number of knots in the function call, then a problem arose if the value of the variable used to specify the number of knots in the function call changes subsequently. 

I wonder if the number of knots or the basis functions used could be stored in the output to prevent this issue? 

The following code reproduces the issue using version 2.14.4 of brms.

x <- rnorm(100)
y <- sin(x/3) + rnorm(100,0,1/4)

dat <- data.frame(x,y)

# Working example 1
library(brms)
fit1 <- brm(y~s(x),data=dat)
predict(fit1,type="response")

# Working example 2
k<-11
fit2 <- brm(y~s(x,bs="tp",k=k))
predict(fit2,type="response")

# Creating the issue
k<-14
predict(fit2,type="response")

"Error in tcrossprod(b, X) : non-conformable arguments"

**Comment by paul-buerkner** [comment]
Thanks! Can mgcv (the package use by brms to prepare splines) handle this case? It seems to me as if we would usually not use non-data variables in formulas taken from the global env and instead always hard-code in the formula, e.g., via `s(x, k = 14)`. I am not sure if there is a fix for it to be honest.

**Comment by tbmurphy** [comment]
mgcv seems to be unaffected by the problem. It's no longer an issue for me now that I know the problem.

The issue first arose with my code when I was trying multiple values of k and then storing the fits. When I went back to the fits later, the predict() command wasn't working sometimes because the k used for a particular fit didn't always match the one in the global environment.

But, as you say, it is a very specific problem. The package is working great for my work anyway now that I have diagnosed the issue. 

Thanks!
